lib/stack_master/aws_driver/s3.rb

```diff
@@ -48,6 +48,31 @@ def upload_files(bucket: nil, prefix: nil, region: nil, files: {})
         end
       end
 
+
+      def download_file(bucket, prefix, region, filename)
+        raise StackMaster::AwsDriver::S3ConfigurationError, 'A bucket must be specified in order to use S3' unless bucket
+
+        s3 = new_s3_client(region: region)
+        StackMaster.stdout.puts "Downloading file from S3"
+
+        downloaded_file = File.join(Dir.tmpdir(), filename)
+        response = s3.get_object(
+          {
+            bucket: bucket,
+            key: "#{prefix}/#{filename}",
+          },
+          target: downloaded_file,
+        ) rescue nil
```

Comment:
Generally speaking we should avoid `rescue nil`, as it catches a lot of Exceptions we don't want to catch, such as a keyboard interrupt.

All s3 sdk errors are a subclass of `Aws::S3::Errors::ServiceError`, so we should instead rescue that exception. We should also do something when that exception occurs to tell the user what happened and exit cleanly.